### ISSUE TITLE
[BUG] Gitea Saver doesn't include hash in final PUT reauest

### ISSUE THREAD

**Issue by Jermolene** [TiddlyWiki/TiddlyWiki5#4823]
> I am trying to use the gitea saver, I have fixed most of the issues but am currently being left with an issue due to the SHA hash not being included with the final put request

(Posted originally on the Google Group - https://groups.google.com/d/msgid/tiddlywiki/c9882547-d736-4682-bec1-9268eb8f089fn%40googlegroups.com?utm_medium=email&utm_source=footer)

![image](https://user-images.githubusercontent.com/174761/90394408-846bfe00-e08a-11ea-9825-6f8072dbd3ef.png)

cc @m42e

**Comment by sunsteam** [comment]
Same problem.  the SHA does in the body , same request is 200 ok in postman ,  why can't work in chrome. 

"tiddlywiki-version 5.1.22"
"Gitea version: 1.13.0+dev-410-g2ef318e6f"


PUT /git/api/v1/repos/sunyuxing/Juno-Wiki/contents/index.html HTTP/1.1
Host: 192.168.1.20
Connection: keep-alive
Content-Length: 3691439
Accept: application/json
X-Requested-With: TiddlyWiki
Authorization: Basic c3VueXV4aW5nOmY2ZDdlMDRmY2ViYThiOGM4YWU0MTg1MmM5NmI1NTg3OWIwNmI1Mjk=
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.135 Safari/537.36
Content-Type: application/json;charset=UTF-8 application/x-www-form-urlencoded; charset=UTF-8
Origin: http://192.168.1.20
Referer: http://192.168.1.20/
Accept-Encoding: gzip, deflate
Accept-Language: zh-CN,zh;q=0.9
Cookie: lang=zh-CN; i_like_gitea=763e6162c2414beb; _csrf=ZueiK7sQIQLgneDon3Oduneyf4Q6MTU5ODkyOTIwNTgyMTQ5MTY1MA

![image](https://user-images.githubusercontent.com/7489390/91791244-37b92300-ec45-11ea-9814-728d22e950f7.png)
![image](https://user-images.githubusercontent.com/7489390/91791373-8d8dcb00-ec45-11ea-915f-523a01b6aa85.png)

**Comment by rmunn** [comment]
@sunsteam Can you post a screenshot of the Network tab in Chrome showing a failing request from the gitea saver? I'm particularly interested in seeing the request headers and body of what the gitea saver is actually sending, to see how they are different from what happens when you send the request in postman.

**Comment by rmunn** [comment]
@sunsteam Never mind, I figured out that your first screenshot was what the Gitea saver sends while the second screenshot was the postman request. It looks like this is a bug that is fixed by https://github.com/Jermolene/TiddlyWiki5/commit/a2796d89ab3dbf4a15b63b0f4623387fdc6acd5b which is in TiddlyWiki 5.1.23-prerelease. Can you check whether 5.1.23-prerelease solves the problem for you?

**Comment by sunsteam** [comment]
@rmunn 
> It looks like this is a bug that is fixed by [a2796d8](https://github.com/Jermolene/TiddlyWiki5/commit/a2796d89ab3dbf4a15b63b0f4623387fdc6acd5b) which is in TiddlyWiki 5.1.23-prerelease. Can you check whether 5.1.23-prerelease solves the problem for you?


Yes, I found this commit and manually fix problem by replace "Content-type" to "Content-Type". Can work.
Where is 5.1.23-prerelease file ?  I didn't see it.

**Comment by sunsteam** [comment]
@rmunn @Jermolene 
also there is another problem, I deploy gitea and tiddlywiki on `192.168.1.20/git` and `192.168.1.20:80` ,  afterI login gitea,  the tiddlywiki put action fail on 401, perhaps cookie conflict.  

I found two way to fix manually
1. clean the cookie 
2. or revise the js, change header `Authorization: Basic c3VueXV4aW5nOmY2ZDdlMDRmY2ViYThiOGM4YWU0MTg1MmM5NmI1NTg3OWIwNmI1Mjk=` to `Authorization: token "the real token"`

can you fix it on next release ?

![image](https://user-images.githubusercontent.com/7489390/91937272-26931380-ed24-11ea-95a6-6737582d8d0f.png)

PUT /git/api/v1/repos/sunyuxing/Juno-Wiki/contents/index.html HTTP/1.1
Host: 192.168.1.20
Connection: keep-alive
Content-Length: 3691319
Accept: application/json
X-Requested-With: TiddlyWiki
Authorization: Basic c3VueXV4aW5nOmY2ZDdlMDRmY2ViYThiOGM4YWU0MTg1MmM5NmI1NTg3OWIwNmI1Mjk=
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.135 Safari/537.36
Content-Type: application/json;charset=UTF-8
Origin: http://192.168.1.20
Referer: http://192.168.1.20/
Accept-Encoding: gzip, deflate
Accept-Language: zh-CN,zh;q=0.9
Cookie: lang=zh-CN; i_like_gitea=10eb754532062a61; _csrf=ajL8eTIWm3N5013RhGz6DwtHIxw6MTU5OTAwOTMwMTE5NjEyMzc2NA

**Comment by rmunn** [comment]
@sunsteam -

> Where is 5.1.23-prerelease file ? I didn't see it.

https://tiddlywiki.com/prerelease/ is the easiest way to get it.

As for your other comment about the cookie, that looks like a different bug; could you open a new issue to track that one?

**Comment by sunsteam** [comment]
#4837  ok.
